src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
-	fi
-
-	# print the sar report to the resources file
-	echo "SAR$FLAGS report" >> $RESOURCES_FILE
-	sar$FLAGS >> $RESOURCES_FILE
-}
-
-# print the output of "apachectl fullstatus" to the resources file
+  local OPTION=$1
+  local LOGFILE=$2
+  local FLAGS=''
+  # check to see if we're going to use any parameters for sar
+  if [[ ${OPTION} == "r" ]]; then
+    FLAGS=" -r"
+  elif [[ ${OPTION} == "q" ]]; then
+    FLAGS=" -q"
+  else
+    FLAGS=""
+  fi
+
+  echo "SAR${FLAGS} report" >> "${LOGFILE}"
+  sar${FLAGS} >> "${LOGFILE}"
+}
+
+# print the output of "apachectl fullstatus" to the specified file
 print_httpd_fullstatus() {
-	echo "Apache Status report" >> $RESOURCES_FILE
-	apachectl fullstatus >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "Apache Status report" >> "${LOGFILE}"
+  apachectl fullstatus >> "${LOGFILE}"
 }
 
-# print the output of "netstat -ntulpae" to the netstat file
+# print the output of "netstat -ntulpae" to the specified file
 print_netstat() {
-	echo "Network connections" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network connections" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT} >> "${LOGFILE}"
 }
 
-# print the output of "netstat -s" to the netstat file
+# print the output of "netstat -s" to the specified file
 print_netstat_sum() {
-	echo "Network traffic summary" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT_SUM >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network traffic summary" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT_SUM} >> "${LOGFILE}"
 }
 
 # print the output of "mysqladmin status" to the mysql file
 print_mysql() {
-        # We only support a single DOTMYDOTCNF for MySQL when running Plesk
-        if [ -f /etc/psa/.psa.shadow ]
-        then
-                echo "MySQL status" >> $MYSQL_FILE
-                mysqladmin -uadmin -p`cat /etc/psa/.psa.shadow` status >> $MYSQL_FILE
-                print_blankline $MYSQL_FILE
-        else
-                for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" )
-                do
-                        if [ -f $cnf ]
-                        then
-                                echo "MySQL $cnf status" >> $MYSQL_FILE
-                                mysqladmin --defaults-file=$cnf status >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        else
-                                echo "MySQL $cnf for status does not exist. Please check DOTMYDOTCNF in /etc/recap" >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        fi
-                done
-        fi
+  local LOGFILE=$1
+  # We only support a single DOTMYDOTCNF for MySQL when running Plesk
+  local PLESK_FILE="/etc/psa/.psa.shadow"
+  if [[ -r "${PLESK_FILE}" ]]; then
+    echo "MySQL status" >> "${LOGFILE}"
+    mysqladmin -uadmin -p$(cat "${PLESK_FILE}") status >> "${LOGFILE}"
+    print_blankline "${LOGFILE}"
+  else
+    for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" ); do
```

Comment:
Not quite sure of this, to me works just fine, see that I'm allowing two/three ways to define the multiple list of configs:

1. White spaces
1. Comma separated values
1. Both

See this example:

Here the testing script, that mimics that for loop:
```bash
[root@gnu ~]# cat x
#!/bin/bash

multi_cnf() {
  multi_confs="$1"
  echo "processing: --${multi_confs}--"
  for cnf in $( sed -e 's/,/ /g' <<<"${multi_confs}" ); do
    if [[ -r "${cnf}" ]]; then
      echo "found: -${cnf}-"
    else
      echo "canot read -${cnf}-"
    fi
  done
  echo
}

# single
multi_cnf "/tmp/my.cnf0"
# multiple spaces
multi_cnf "/tmp/my.cnf0 /tmp/my.cnf1"
multi_cnf "/tmp/my.cnf0 /tmp/my.cnf1 /tmp/my.cnf2 /tmp/my_none"
multi_cnf " /tmp/my.cnf0  /tmp/my.cnf1   /tmp/my.cnf2   /tmp/my_none"
# multiple commas
multi_cnf "/tmp/my.cnf0,/tmp/my.cnf1"
multi_cnf "/tmp/my.cnf0,/tmp/my.cnf1,/tmp/my.cnf2,/tmp/my_none"
# multiple spaces and commas
multi_cnf " /tmp/my.cnf0, /tmp/my.cnf1,  /tmp/my.cnf2   /tmp/my.cnf3  ,/tmp/_my_none"
```

Creating some empty files
```bash
[root@gnu ~]# touch /tmp/my.cnf{0,1,2,3}
```

Now testing:
```bash
[root@gnu ~]# bash x
processing: --/tmp/my.cnf0--
found: -/tmp/my.cnf0-

processing: --/tmp/my.cnf0 /tmp/my.cnf1--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-

processing: --/tmp/my.cnf0 /tmp/my.cnf1 /tmp/my.cnf2 /tmp/my_none--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-
found: -/tmp/my.cnf2-
cannot read -/tmp/my_none-

processing: -- /tmp/my.cnf0  /tmp/my.cnf1   /tmp/my.cnf2   /tmp/my_none--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-
found: -/tmp/my.cnf2-
cannot read -/tmp/my_none-

processing: --/tmp/my.cnf0,/tmp/my.cnf1--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-

processing: --/tmp/my.cnf0,/tmp/my.cnf1,/tmp/my.cnf2,/tmp/my_none--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-
found: -/tmp/my.cnf2-
cannot read -/tmp/my_none-

processing: -- /tmp/my.cnf0, /tmp/my.cnf1,  /tmp/my.cnf2   /tmp/my.cnf3  ,/tmp/_my_none--
found: -/tmp/my.cnf0-
found: -/tmp/my.cnf1-
found: -/tmp/my.cnf2-
found: -/tmp/my.cnf3-
cannot read -/tmp/_my_none-
```